ddtrace/tracer/option.go

```diff
@@ -297,6 +291,98 @@ type config struct {
 	logDirectory string
 }
 
+var (
+	errOutOfRange = errors.New("value out of range")
+)
+
+var (
+	enabled = knobs.Register(&knobs.Definition[bool]{
+		Default: true,
+		EnvVars: []knobs.EnvVar{
+			{
+				Key: "DD_TRACE_ENABLED",
+			},
+			{
+				Key:       "OTEL_TRACES_EXPORTER",
+				Transform: mapEnabled,
+			},
+		},
+		Parse: func(s string) (bool, error) {
+			if s == "" {
+				// Short-circuit if the value is empty, to avoid parsing errors.
+				return true, nil
+			}
+			return strconv.ParseBool(s)
+		},
+	})
+
+	// globalSampleRate holds sample rate read from environment variables.
+	globalSampleRate = knobs.Register(&knobs.Definition[float64]{
+		Default: math.NaN(),
+
+		EnvVars: []knobs.EnvVar{
+			{
+				Key: "DD_TRACE_SAMPLE_RATE",
+			},
+			{
+				Key:       "OTEL_TRACES_SAMPLER",
+				Transform: mapSampleRate,
+			},
+		},
+		Resolve: func(environ map[string]string, decision string) (string, error) {
```

Comment:
@mtoffl01 As you already saw, I took the liberty to create a `Resolve` function implementing what we discussed offline. This `Resolve` allows to do extra validations in the whole context, but also potentially overriding what environment variable should be used.